pkg/client/core/api/rest/domain.go

```diff
@@ -12,6 +15,10 @@ type domainAPI struct {
 	client *HTTPClient
 }
 
+func (a *domainAPI) DeleteHostedZoneRecords(provider v1alpha1.CloudProvider, hostedZoneID string) (*route.ChangeResourceRecordSetsOutput, error) {
```

Comment:
Strictly speaking we should implement this delete functionality on the server-side, and then invoke the corresponding REST-API from this side